src/irmin-pack/dict.ml

```diff
@@ -79,21 +79,25 @@ module Make (IO : IO.S) : S = struct
     in
     (aux [@tailcall]) (Hashtbl.length t.cache) 0
 
-  (** if RW was cleared we have to refill the buffers entirely. In both cases
-      offset needs to be updated, as the offset is used in the refill. *)
-  let sync_offset t =
-    let former_log_offset = IO.offset t.io in
-    let log_offset = IO.force_offset t.io in
-    let former_generation = IO.generation t.io in
-    let generation = IO.force_generation t.io in
-    if former_generation <> generation then (
+  let sync_offset t ~force_refill =
+    if force_refill then (
+      (* RW was cleared, so we have to reopen file and refill the hashtables
+         entirely *)
```

Comment:
I'm confused about this comment. `force_refill` is passed by the user of `Dict`, so this feels like the wrong place to conjecture about _why_ `force_refill` might have been passed. Relatedly, I think it'd be good to document what this parameter does in the `.mli` file.

Reply:
I removed the comment and added this [one](https://github.com/mirage/irmin/pull/1015/files#diff-20f45fef7076110026f9330f86fa4723R26-R29) instead.

Reply:
Thanks :)